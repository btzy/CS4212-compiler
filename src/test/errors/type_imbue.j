class Main {

Void main(Int i, Int a, Int b,Int d){
    Int j;
    i = "abc";
    j = true;
    (new A()).a = "ghi";
}
}
class A {
    Int a;
}