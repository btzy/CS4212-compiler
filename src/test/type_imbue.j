class Main {

Void main(Int i, Int a, Int b,Int d){
    Int j;
    //i = "abc";
    //j = "def";
    (new A()).a = "ghi";
}
}
class A {
    Int a;
}