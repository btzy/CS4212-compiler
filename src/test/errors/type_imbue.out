Error: Expected expression to have type "Int", but got type "String" instead ( line 5 col 9 - line 5 col 13 ) :
5 |     i = "abc";
  |         ^^^^^
Note: Variable "i" was declared as type "Int" here ( line 3 col 11 - line 3 col 15 ) :
3 | Void main(Int i, Int a, Int b,Int d){
  |           ^^^^^
Error: Expected expression to have type "Int", but got type "Bool" instead ( line 6 col 9 - line 6 col 12 ) :
6 |     j = true;
  |         ^^^^
Note: Variable "j" was declared as type "Int" here ( line 4 col 5 - line 4 col 9 ) :
4 |     Int j;
  |     ^^^^^
Error: Expected expression to have type "Int", but got type "String" instead ( line 7 col 19 - line 7 col 23 ) :
7 |     (new A()).a = "ghi";
  |                   ^^^^^
Note: Variable "a" was declared as type "Int" here ( line 11 col 5 - line 11 col 9 ) :
11 |     Int a;
   |     ^^^^^
Semantic checking aborted due to above errors.
